src/Trip.js

```diff
@@ -0,0 +1,52 @@
+
+class Trip {
+  constructor(currentUserId, dataForRequestedTrip, allTripData, destinationData) {
+    this.currentUserId = currentUserId
+    this.tripData = allTripData
+    this.id = this.findID()
+    this.dataForTrip = dataForRequestedTrip;
+    this.travelers = dataForRequestedTrip.travelers
+    this.status = 'pending'
+    this.date = dataForRequestedTrip.startDate
+    this.destinationData = destinationData;
+    this.estimatedCost = this.calculateEstimatedCost()
+  }
+  calculateEstimatedCost() {
+    let acc = 0
+    let correctDestination = this.destinationData.find(destination => {
+      return this.dataForTrip.destination === destination.destination
+    })
+    acc += correctDestination.estimatedLodgingCostPerDay * 4
+    acc += (this.dataForTrip.travelers * correctDestination.estimatedFlightCostPerPerson)
+    return acc
+  }
+  findID () {
+    return this.tripData.reduce((acc, trip) => {
+      if (acc === trip.id || acc === 0) {
+        acc += this.findUniqueValue(trip.id, acc)
+      } else {
+        return acc
+      }
+      return acc
+    }, 0)
+  }
+  findUniqueValue(trip, value) {
+    value += 1
+    if (trip === value || value === 0 ) {
```

Comment:
I'm not certain, but I don't think you need this function right now. 

I suppose you could add other functions later that need it, but the only place you're using it, you already know `acc === trip.id`. This whole function could be reduced to `return trip.id + 1`. Since you are logically guaranteed (in the one place you've invoked it so far) to just add 1, why not just delete the function, and replace the call with `acc += acc + 1` ? 

Also, is there a reason you're roughly doubling your ID with every match? I haven't checked, but unless your `this.tripData` is sorted in increasing id order, there are still chances that you might have a ID collision.

I recommend either `sort`ing first or perhaps rethinking this algorithm. (I'm also not positive but since this algorithm hits some pretty big numbers pretty fast, it might be worthwhile to check if there is an upper bound to what a valid ID might be.)

Reply:
I totally assumed that it was in increasing order, now that you bring it up I definitely see how it doesn't make since without sorting first!